File: backend/src/main/java/com/construction/client/entity/AppRole.java
package com.construction.client.entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "AppRoles")
public class AppRole {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "RoleID")
    private Integer roleId;

    @Column(name = "RoleName", nullable = false)
    private String roleName;

    @Column(name = "Description")
    private String description;
}

File: frontend/lib/models/app_role.dart
class AppRole {
  final int id;
  final String name;
  final String description;

  AppRole({
    required this.id,
    required this.name,
    required this.description,
  });

  factory AppRole.fromJson(Map<String, dynamic> json) {
    return AppRole(
      id: json['roleId'] ?? json['id'] ?? 0,
      name: json['roleName'] ?? json['name'] ?? '',
      description: json['description'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'roleId': id,
      'roleName': name,
      'description': description,
    };
  }
}
